src/submission/validation-clinical/donor.ts

```diff
@@ -138,81 +145,126 @@ const crossFileValidator = async (
+            lostToFollowUpClinicalInfo.specimen_duration,
+        ) || 0;
+
+      const lostToFollowUpInterval = lostToFollowUpStartInterval + lostToFollowUpDuration;
+
+      // Collects all Entity Records w/ Treatment Intervals after Lost to Follow Up
+      const invalidClinicalIntervalRecords = [...treatments, ...specimens]
+        .filter(entityRecord => {
+          const { clinicalInfo } = entityRecord;
+          const { treatment_start_interval, treatment_duration } = clinicalInfo;
+          const { specimen_acquisition_interval, specimen_duration } = clinicalInfo;
+
+          const treatmentInterval =
+            typeof treatment_start_interval === 'number'
+              ? treatment_start_interval + (Number(treatment_duration) || 0)
+              : 0;
+
+          const specimenInterval =
+            typeof specimen_acquisition_interval === 'number'
+              ? specimen_acquisition_interval + (Number(specimen_duration) || 0)
+              : 0;
+
+          return (
+            treatmentInterval > lostToFollowUpInterval || specimenInterval > lostToFollowUpInterval
+          );
+        })
+        .map(record => record.clinicalInfo);
+
+      // Collects all Records w/ Follow Up Intervals greater than Lost to Follow Up
+      const invalidFollowUpIntervalRecords = followUps
+        .filter(entityRecord => {
+          const clinicalInfo = entityRecord.clinicalInfo;
+          const { interval_of_followup } = clinicalInfo;
+          const intervalOfFollowUp =
+            typeof interval_of_followup === 'number' ? interval_of_followup : 0;
+          return intervalOfFollowUp > lostToFollowUpInterval;
+        })
+        .map(record => record.clinicalInfo);
+
+      const invalidRecords = [...invalidClinicalIntervalRecords, ...invalidFollowUpIntervalRecords];
+
+      for (const invalidClinicalInfo of invalidRecords) {
+        const {
+          submitter_treatment_id,
+          submitter_specimen_id,
+          submitter_follow_up_id,
+        } = invalidClinicalInfo;
+        const treatment_id =
+          submitter_treatment_id || submitter_specimen_id || submitter_follow_up_id;
+
+        errors.push(
+          utils.buildSubmissionError(
+            submittedDonorRecord,
+            DataValidationErrors.INVALID_SUBMISSION_AFTER_LOST_TO_FOLLOW_UP,
+            DonorFieldsEnum.lost_to_followup_after_clinical_event_id,
+            {
+              lost_to_followup_after_clinical_event_id,
+              interval_of_followup: lostToFollowUpInterval,
+              treatment_id,
+            },
+          ),
+        );
+      }
 
-        const lostToFollowUpPrimaryDiagnosisMatch = primaryDiagnoses?.find(
-          primaryDiagnosisRecord =>
-            primaryDiagnosisRecord?.clinicalInfo?.submitter_primary_diagnosis_id ===
-            lostToFollowUpDiagnosisId,
-        )?.clinicalInfo;
+      // Filter Primary Diagnosis records with Age at Diagnosis greater than age at Lost To Follow Up
```

Comment:
Primary Diagnosis contains `age` but not `interval` data (years vs days) and is checked separately